roxie/udplib/udptrs.cpp

```diff
@@ -204,7 +204,15 @@ class UdpReceiverEntry : public IUdpReceiverEntry
             {
                 if (encrypted)
                 {
-                    aesEncrypt(key, sizeof(key), buffer->data, length, encryptBuffer.clear());
+                    encryptBuffer.clear();
+                    encryptBuffer.append(sizeof(UdpPacketHeader), header);    // We don't encrypt the header
+                    length -= sizeof(UdpPacketHeader);
+                    const char *data = buffer->data + sizeof(UdpPacketHeader);
+                    aesEncrypt(key, sizeof(key), data, length, encryptBuffer);
+                    header->length = encryptBuffer.length();
+                    encryptBuffer.writeDirect(0, sizeof(UdpPacketHeader), header);   // Only really need length updating
+                    assert(length <= DATA_PAYLOAD);
+                    DBGLOG("ENCRYPT: Writing %u bytes to data socket", encryptBuffer.length());
```

Comment:
Excessive tracing...

Reply:
This looks good, but I am missing where the extra possible 16 bytes of encryptBuffer are allocated ?

Reply:
ok, I see inside MemoryBuffer &aesEncrypt(const void *key, size_t keylen, const void *input, size_t inlen, MemoryBuffer &output) is:
output.reserveTruncate(truncInLen + 16));